filterable-file-log/serializer.lua

```diff
@@ -60,11 +64,46 @@ function _M.serialize(ngx, filters)
     service = ngx.ctx.service,
     api = ngx.ctx.api,
     consumer = ngx.ctx.authenticated_consumer,
+    jwt_claims = jwt_claims,
     client_ip = ngx.var.remote_addr,
     started_at = ngx.req.start_time() * 1000
   }
 end
 
+function get_jwt_claims(headers)
+  if not headers['authorization'] then
+    return nil
+  end
+
+  local token = string.match(headers['authorization'], 'Bearer (.+)')
+  if not token then
+    return nil
+  end
+
+  -- The JWT has three .-delimited parts: header, payload, and signature
+  -- The payload has the claims we're interested in
+  local encoded_claims = stringx.split(token, '.')[2]
+  if not encoded_claims then
+    return nil
+  end
+
+  local claims = ngx.decode_base64(encoded_claims)
+  local parsed_claims, err = cjson.decode(claims)
+  if err then
+    return nil
+  end
```

Comment:
What does `decode_base64` return if the content is not truly base64? Is it possible that `claims` is nil here?

Reply:
It returns `nil` (https://github.com/openresty/lua-nginx-module#ngxdecode_base64) so yes, it's possible `claims` is `nil`. That's not really a problem because `cjson.decode` will safely return an error in that case and all we want to do is return `nil`. I'll go ahead and explicitly handle that case just to make this clearer.